storybook/theme-switcher-addon/themeManager.ts

```diff
@@ -0,0 +1,25 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import {
+  defaultTheme,
+  heartTheme,
+  ThemeManager,
+  zenTheme,
+} from "@kaizen/design-tokens"
+import { THEME_KEY_STORE_KEY } from "./constants"
+export const themeOfKey = (themeKey: string) => {
+  switch (themeKey) {
+    case "heart":
+      return heartTheme
+    default:
+      return zenTheme
+  }
+}
+
+export const getInitialTheme = () =>
+  themeOfKey(
+    window.location.search.match(/(\?|\&)theme=(zen|heart)/)?.[2] ||
```

Comment:
I would think that the logic would work something like this (feel free to challenge, of course)

```
If query param exists for theme then 
   set that theme in local storage (overriding existing value if it exists)
   set that theme's variables on root of document
else
   if theme is defined in local storage then
      use local storage theme
   else
      use default theme (zen)
```

I've opened a new tab with the theme, and tried to override the theme with a query parameter and it seems the only way to change the theme is to open another tab and change the theme there.

Reply:
@ActuallyACat i did some digging because i could swear my experience of the logic was exactly as you described.
Turns out the useEffect hook that updates localStorage isn't run when viewing the canvas/viewOnly tab, because the addons aren't loaded!
Amending this now.